server/storage/mvcc/watcher.go

```diff
@@ -22,9 +22,14 @@ import (
 	"go.etcd.io/etcd/api/v3/mvccpb"
 )
 
-// AutoWatchID is the watcher ID passed in WatchStream.Watch when no
-// user-provided ID is available. If pass, an ID will automatically be assigned.
-const AutoWatchID WatchID = 0
+const (
+	// InvalidWatchID represents an invalid watch ID and prevents duplication with an existing watch.
+	InvalidWatchID WatchID = -1
+
+	// AutoWatchID is the watcher ID passed in WatchStream.Watch when no
+	// user-provided ID is available. If pass, an ID will automatically be assigned.
+	AutoWatchID WatchID = 0
```

Comment:
I suggest to remove the duplicated definition, and reuse the definitions in clientv3.